recipes/scip/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+    settings = "os", "arch", "compiler", "build_type"
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_boost": [True, False],
+        "with_gmp": [True, False],
+        "with_tpi": ["none", "omp", "tny"],
+        "with_sym": ["bliss", "none"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_boost": True,
+        "with_gmp": True,
+        "with_tpi": "none",
+        "with_sym": "bliss",
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+        if self.options.shared and self.options.with_sym == "bliss":
+            raise ConanInvalidConfiguration("Bliss is not supported in shared mode.")
+        comp = self.settings.compiler
+        if self.options.with_sym == "bliss" and comp == 'clang' and comp.libcxx and comp.libcxx == 'libc++':
+            raise ConanInvalidConfiguration("Bliss does not support libc++.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("soplex/6.0.3")
+
+    def configure(self):
+        self.options["soplex"].with_boost = self.options.with_boost
+        self.options["soplex"].with_gmp = self.options.with_gmp
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    @staticmethod
+    def _to_cmake(*arrays):
+        return ";".join(item.replace("\\", "/") for sublist in arrays for item in sublist)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SHARED"] = self.options.shared
+        tc.variables["READLINE"] = False  # required for interactive stuff
+        tc.variables["GMP"] = self.options.with_gmp
+        tc.variables["BOOST"] = self.options.with_boost
+        tc.variables["TPI"] = self.options.with_tpi
```

Comment:
You should declared explicitly each dependency in this recipe, even if soplex is consuming the same dependencies, the dependencies graph should not get a ride from soplex.

Reply:
I don't understand your comment. What am I doing wrong? What should I change?

Reply:
with_boost and with_gmp. It's configuring to build with Boost, but boost and gmp are not listed as requirements in this reacipe. Every dependency should be listed in the recipe. 

Sometimes a dependencies already fills the gap, importing a dependency that's needed by both packages, but is a fragile practice.

Reply:
looking into the upstream code, I don't see boost usage:

```
 /tmp/scip-803  % grep BOOST -r -i
./Makefile:SOFTLINKS    +=    $(LIBDIR)/include/boost
./Makefile:LPIINSTMSG    +=    "\n  -> \"boost\" is the path to the boost include folder\n"
./INSTALL.md:apt-get install wget cmake g++ m4 xz-utils libgmp-dev unzip zlib1g-dev libboost-program-options-dev libboost-serialization-dev libboost-regex-dev libboost-iostreams-dev libtbb-dev libreadline-dev pkg-config git liblapack-dev libgsl-dev flex bison libcliquer-dev gfortran file dpkg-dev libopenblas-dev rpm
./INSTALL.md:| `BOOST_ROOT`           | `<custom/path/to/.../boost>`       | --                         | hint for location of boost                                         |
./doc/xternal.c: *          to change many parameters at once and boost the performance.</li>
./src/scip/intervalarith.c: * taken from boost::numeric::interval_lib::constants
./src/scip/intervalarith.c:   /* this implementation follows boost::numeric::cos
./src/cppad/core/testvector.hpp:$head boost::numeric::ublas::vector$$
./src/cppad/core/testvector.hpp:you specify $icode cppad_testvector$$ to be $code boost$$,
./src/cppad/core/testvector.hpp:$code CPPAD_BOOSTVECTOR$$ will be true.
./src/cppad/core/testvector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/testvector.hpp:# include <boost/numeric/ublas/vector.hpp>
./src/cppad/core/testvector.hpp:# define CPPAD_TESTVECTOR(Scalar) boost::numeric::ublas::vector< Scalar >
./src/cppad/core/testvector.hpp:In this case CppAD will use this boost vector for
./src/cppad/core/test_vector.hpp:The include path for boost is not defined in the Windows project files.
./src/cppad/core/test_vector.hpp:of $code CPPAD_BOOSTVECTOR$$:
./src/cppad/core/test_vector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:# undef  CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:# define CPPAD_BOOSTVECTOR 0
./src/cppad/core/test_vector.hpp:$head boost::numeric::ublas::vector$$
./src/cppad/core/test_vector.hpp:If you specify a value for $icode boost_dir$$ on the configure
./src/cppad/core/test_vector.hpp:$code CPPAD_BOOSTVECTOR$$ is true
./src/cppad/core/test_vector.hpp:# if CPPAD_BOOSTVECTOR
./src/cppad/core/test_vector.hpp:# include <boost/numeric/ublas/vector.hpp>
./src/cppad/core/test_vector.hpp:# define CPPAD_TEST_VECTOR boost::numeric::ublas::vector
./src/cppad/core/test_vector.hpp:Specifying $icode boost_dir$$ will increase the amount of
./src/cppad/core/undef.hpp:# undef CPPAD_BOOSTVECTOR
./src/cppad/configure.hpp:\def CPPAD_BOOSTVECTOR
./src/cppad/configure.hpp:we are using boost vector for CPPAD_TESTVECTOR.
./src/cppad/configure.hpp:we are not using boost vector for CPPAD_TESTVECTOR.
./src/cppad/configure.hpp:# define CPPAD_BOOSTVECTOR 0
./src/amplmp/include/mp/os.h:// This class provides a subset of boost filesystem's path API.
```

Is cppad built and packaged together? If not, I don't see a need for keeping the option `with_boost`.

Reply:
cppad is used https://github.com/scipopt/scip/blob/4de2d40a3f7998540291a6237d14923d9a201eb9/CMakeLists.txt#L290 if the default is not changed https://github.com/scipopt/scip/blob/4de2d40a3f7998540291a6237d14923d9a201eb9/CMakeLists.txt#L102

Reply:
the dependency soplex would take care of gmp and boost, also in the soplex package adds these dependencies only when the parameters are set https://github.com/conan-io/conan-center-index/blob/master/recipes/soplex/all/conanfile.py#L79

should I add the dependencies on boost and gmp independently of the parameters the user sets?

Reply:
okay, then you need to include boost as a package dependency in this recipe:

```
def requirements(self):
    self.requires("1.82.0")
```

same for `gmp`

Reply:
OK, added